storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -3108,17 +3108,22 @@ static void xtrabackup_init_datasinks(void) {
                        : DS_TYPE_COMPRESS_QUICKLZ);
     xtrabackup_add_datasink(ds);
     ds_set_pipe(ds, ds_data);
-    if (ds_data != ds_redo) {
+
+    if (srv_redo_log_encrypt)
```

Comment:
What if the variable is turned off after encryption is turned for a while? Shouldn't we check the redo log header to be sure?

Reply:
it is an overhead. we have fixed the code of compression.  even if it compressing an encrypted log file it won't cause any trouble.

Reply:
I think, its the other way round. Wrong detection of encrypted redo log would lead un-necessary attempts to compress it.

Since we parse redo log header, we should detect if redo is encrypted or not based on it. This is accurate.